source/io/module_javaScriptInvoke.js

```diff
@@ -285,49 +285,62 @@
 
         var updateReturnValue = function (
             functionName,
-            returnPointer,
-            returnValue,
+            returnTypeName,
+            returnValuePointer,
+            returnUserValue,
             errorStatusPointer,
             errorCodePointer,
             errorSourcePointer) {
-            if (!isValidJavaScriptReturnType(returnValue)) {
+            if (!isValidJavaScriptReturnType(returnUserValue)) {
                 var code = ERRORS.kNIUnsupportedJavaScriptReturnTypeInJavaScriptInvoke.CODE;
                 var source = ERRORS.kNIUnsupportedJavaScriptReturnTypeInJavaScriptInvoke.MESSAGE + '\'' + functionName + '\'.';
                 Module.coreHelpers.mergeErrors(true, code, source, errorStatusPointer, errorCodePointer, errorSourcePointer);
                 return;
             }
 
-            var returnValueIndex = 0;
-            var returnTypeName = getParameterTypeString(returnPointer, returnValueIndex);
-
             if (returnTypeName === 'StaticTypeAndData') {
                 // User doesn't want return value. If we're passing '*' for the return in VIA code, we get StaticTypeAndData
                 return;
             }
 
-            var returnValuePointer = JavaScriptInvoke_GetParameterPointer(returnPointer, 0);
-            if (returnTypeName === 'Array') {
-                returnTypeName += getArrayElementTypeString(returnValuePointer);
-            }
-
             var typeConfig = typeFunctions[returnTypeName];
 
             if (typeConfig === undefined) {
                 var source2 = ERRORS.kNIUnsupportedLabVIEWReturnTypeInJavaScriptInvoke.MESSAGE;
                 var code2 = ERRORS.kNIUnsupportedLabVIEWReturnTypeInJavaScriptInvoke.CODE;
                 Module.coreHelpers.mergeErrors(true, code2, source2, errorStatusPointer, errorCodePointer, errorSourcePointer);
                 return;
-            } else if (!typeConfig.isValidReturnType(returnValue)) {
+            } else if (!typeConfig.isValidReturnType(returnUserValue)) {
                 var source3 = ERRORS.kNITypeMismatchForReturnTypeInJavaScriptInvoke.MESSAGE;
                 var code3 = ERRORS.kNITypeMismatchForReturnTypeInJavaScriptInvoke.CODE;
                 Module.coreHelpers.mergeErrors(true, code3, source3, errorStatusPointer, errorCodePointer, errorSourcePointer);
                 return;
             }
 
-            typeConfig.writer(returnValuePointer, returnValue);
+            typeConfig.writer(returnValuePointer, returnUserValue);
+        };
+
+        var generateCompletionCallback = function (occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer) {
+            var completionCallbackInvoked = false;
+            return function (returnValue) {
```

Comment:
It would be nice to name this function so it has a good name in stack traces

Reply:
If eslint gets mad at you just do:
var completionCallback = function () {
};
return completionCallback;

In ES6 the binding name to an anonymous function expression becomes the functions names